Lib/test/test_code.py

```diff
@@ -428,6 +436,19 @@ def func():
             self.assertIsNone(line)
             self.assertEqual(end_line, new_code.co_firstlineno + 1)
 
+    def test_large_lnotab(self):
+        d = {}
+        lines = (
+            ["def f():"] +
+            [""] * (1 << 17) +
+            ["    pass"] * (1 << 17)
+        )
+        source = "\n".join(lines)
+        exec(source, d)
+        code = d["f"].__code__
+
+        self.assertEqual(len(code.co_lnotab), 264208)
```

Comment:
We could also make this a proper test for the values if you want:

```suggestion
        expected = 1032 * [0, 127] + [0, 9] + ((1 << 17) - 1) * [2, 1]
        self.assertEqual(list(code.co_lnotab), expected)
```